### PR TITLE
Enable deleting connections

### LINKED ISSUE #57: implement connection deletion
connections can be created now, but not yet deleted

### PR DESCRIPTION
**Pull request by sjvrijn**
Fixes #57

Uses the same interaction as creating the statements in the first place: right click statement, click 'delete connection', click other statement to delete connection with it.

Changes summary:

- add deleteConnection function
- separate drawConnection as function
- use statement.id instead of rowindex

### CODE REVIEW

File: includes/chart.js [hunk 1]
```diff
@@ -542,52 +545,85 @@ function getParentTransform(node) {
...
-});
+
+    // Set the attributes with the updated values
+    line.setAttribute("data-start-row-id", startRowIdNum);
+    line.setAttribute("data-end-row-id", destinationRowIdNum);
+    line.setAttribute("id", "connector_" + startRowIdNum + "-" + destinationRowIdNum);
+    line.setAttribute("start-shape-id_", startShapeId);
+    line.setAttribute("end-shape-id_", destinationShapeId);
+
+    console.log(line.getAttribute('id'));
```
Comment (ermannolocascio) [inline includes/chart.js:594]: I would remove or comment (//) the console.log once the code is completed.

**Review by ermannolocascio — COMMENTED** [review]
For some reason the delete connection is not working on my local. Better to double check as soon as we can.

File: includes/chart.js (hunk 1 above)
Comment (sjvrijn) [reply includes/chart.js:594]: Ah yes, that `console.log` is left from debugging, good catch!

**Comment by sjvrijn** [comment]
That's odd, are you sure you're on the right branch? One minor thing I realized you have to pay attention to is to click the shape, not the text in the shape as that doesn't pass the click event through to it's parent (yet).

Should I make a separate PR for that, or include it in this one?

**Comment by ermannolocascio** [comment]
I have tried again: apparently is not working on my local. We can chekck it tmr in person.

**Comment by ermannolocascio** [comment]
UPDATE: Just realized how it works. Better to add a comment on the help page to clarify how the deleting connection works.

**Review by ermannolocascio — APPROVED** [review]

**Comment by sjvrijn** [comment]
Thanks for checking and approving @ermannolocascio , happy to hear you got it to work in the end. How could I have changed the PR description to make more clear how the functionality is supposed to work?

**Pull request merged**